File: additive_animations/src/main/java/at/wirecube/additiveanimations/additive_animator/animation_set/SingleAnimationAction.java
package at.wirecube.additiveanimations.additive_animator.animation_set;

import android.animation.TypeEvaluator;
import android.util.Property;

import java.util.ArrayList;
import java.util.List;

public class SingleAnimationAction<T> implements AnimationAction<T> {
    private final List<AnimationAction.Animation<T>> mAnimations = new ArrayList<>();

    @Override
    public List<AnimationAction.Animation<T>> getAnimations() {
        return mAnimations;
    }

    public SingleAnimationAction(Property<T, Float> property, float target) {
        mAnimations.add(new AnimationAction.Animation<>(property, target));
    }

    public SingleAnimationAction(Property<T, Float> property, float target, TypeEvaluator<Float> evaluator) {
        mAnimations.add(new AnimationAction.Animation<T>(property, target, evaluator));
    }

}

File: additive_animations/src/main/java/at/wirecube/additiveanimations/additive_animator/animation_set/SingleAnimationAction.kt
package at.wirecube.additiveanimations.additive_animator.animation_set

import android.animation.TypeEvaluator
import android.util.Property

class SingleAnimationAction<T> : AnimationAction<T> {

    private val animations: MutableList<AnimationAction.Animation<T>> = mutableListOf()

    constructor(property: Property<T, Float>, target: Float) {
        animations.add(AnimationAction.Animation(property, target))
    }

    constructor(property: Property<T, Float>, target: Float, evaluator: TypeEvaluator<Float>) {
        animations.add(AnimationAction.Animation(property, target, evaluator))
    }

    override fun getAnimations(): List<AnimationAction.Animation<T>> = animations
}

